pkg/addons/default/coredns_test.go

```diff
@@ -106,41 +109,62 @@ var _ = Describe("default addons - coredns", func() {
 
 	})
 
-	Describe("can update coredns", func() {
-		var (
-			clientSet *fake.Clientset
-		)
-
-		check := func(imageTag string) {
-			coreDNS, err := clientSet.AppsV1().Deployments(metav1.NamespaceSystem).Get(CoreDNS, metav1.GetOptions{})
+	Context("[1.11 –> 1.12] can update coredns", func() {
 
-			Expect(err).ToNot(HaveOccurred())
-			Expect(coreDNS).ToNot(BeNil())
-			Expect(coreDNS.Spec.Template.Spec.Containers).To(HaveLen(1))
-
-			Expect(coreDNS.Spec.Template.Spec.Containers[0].Image).To(
-				Equal("602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/coredns:" + imageTag),
-			)
-		}
-
-		BeforeEach(func() {
-			clientSet, _ = testutils.NewFakeClientSetWithSamples("testdata/sample-1.11.json")
-		})
+		loadSample("1.11", 10)
 
 		It("can load 1.11 sample", func() {
-			check("v1.1.3")
+			checkCoreDNSImage(rawClient, "eu-west-1", "v1.1.3")
 		})
 
 		It("can update based to latest version", func() {
-			_, err := UpdateCoreDNSImageTag(clientSet, false)
+			_, err := UpdateCoreDNS(rawClient, "eu-west-2", "1.12.x", false)
```

Comment:
@martina-if what do you make of this test? It checks as much as currently possible with this fake client of ours...

Reply:
I suppose we could add tests with other kind of pre-conditions that would result in errors, if you think that's important?

Reply:
The problem I see is that if I break the logic in that function this test won't caught it because it only checks that there is no error returned. I prefer tests that tell you that you actually broke some piece of logic. Maybe it's hard to test with how it looks now. Perhaps it needs to be split in smaller pieces that can be tested. Or test that we call the client with the right information?

Reply:
See below, the test looks at what requests were made - does that seem sufficient?

Reply:
ok, I don't see a better way right now to test this :+1: